mobile/src/main/java/org/openhab/habdroid/model/SitemapImpl.java

```diff
@@ -0,0 +1,89 @@
+package org.openhab.habdroid.model;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@EqualsAndHashCode
+public class SitemapImpl implements Sitemap {
+    private static final String TAG = SitemapImpl.class.getSimpleName();
+    public static final Parcelable.Creator<Sitemap> CREATOR = new Parcelable.Creator<Sitemap>() {
+        public Sitemap createFromParcel(Parcel in) {
+            ObjectMapper mapper = new ObjectMapper();
+            String jsonData = in.readString();
+
+            try {
+                return mapper.readValue(jsonData, SitemapImpl.class);
```

Comment:
I'm unsure: If this fails, should we (until one of the next play store updates) have a fallback to the "old" parcel reading and setting of values? I'm not sure, if svaed parcels are saved between version updates?

Reply:
They are not. From [the docs](https://developer.android.com/reference/android/os/Parcel.html):

> Parcel is not a general-purpose serialization mechanism. This class (and the corresponding Parcelable API for placing arbitrary objects into a Parcel) is designed as a high-performance IPC transport. As such, it is not appropriate to place any Parcel data in to persistent storage: changes in the underlying implementation of any of the data in the Parcel can render older data unreadable.

Reply:
So, we're fine here, right? :)

Reply:
Yes.